src/test/components/SampleTooltipContents.test.js

```diff
@@ -6,48 +6,121 @@
 
 import React from 'react';
 import { Provider } from 'react-redux';
-import { SampleTooltipContents } from 'firefox-profiler/components/shared/SampleTooltipContents';
-import { render, screen } from '@testing-library/react';
-import {
-  getCategories,
-  getThreadSelectorsFromThreadsKey,
-  getMaxThreadCPUDelta,
-  getSampleUnits,
-  getProfileInterval,
-} from 'firefox-profiler/selectors';
+import { TimelineTrackThread } from 'firefox-profiler/components/timeline/TrackThread';
+import { render, screen, fireEvent } from '@testing-library/react';
 import { ensureExists } from 'firefox-profiler/utils/flow';
 import { storeWithProfile } from '../fixtures/stores';
 import {
   getProfileFromTextSamples,
   addCpuUsageValues,
 } from '../fixtures/profiles/processed-profile';
-
-import type { Profile } from 'firefox-profiler/types';
+import {
+  getMouseEvent,
+  getBoundingBox,
+  addRootOverlayElement,
+  removeRootOverlayElement,
+} from '../fixtures/utils';
+import { autoMockCanvasContext } from '../fixtures/mocks/canvas-context';
+
+import type {
+  Profile,
+  IndexIntoSamplesTable,
+  CssPixels,
+  ThreadCPUDeltaUnit,
+} from 'firefox-profiler/types';
+
+// The following constants determine the size of the drawn graph.
+const SAMPLE_COUNT = 4;
+const PIXELS_PER_SAMPLE = 40;
+const GRAPH_WIDTH = Math.floor(PIXELS_PER_SAMPLE * SAMPLE_COUNT);
+const GRAPH_HEIGHT = 10;
```

Comment:
Since I'm not rendering the activity graph too, we had to add these here as well. They are mostly copy paste from other tests with changed values for convenience. `getSamplesPixelPosition` is different though, because we need to point out "before" and "after" parts of the sample.